tooling/bundler/src/bundle/linux/debian.rs

```diff
@@ -380,6 +382,16 @@ fn create_tar_from_dir<P: AsRef<Path>, W: Write>(src_dir: P, dest_file: W) -> cr
       header.set_metadata(&stat);
       header.set_uid(0);
       header.set_gid(0);
+
+      // It extracts the MSB in octal permissions.
+      let user_permission = (stat.mode()%512)/64;
+      // Check if file is executable, and set permissions accordingly.
+      if user_permission%2 == 1{
```

Comment:
Using bitwise operator might be easier to read:
```suggestion
      // Check if file is user-executable, and set permissions accordingly.
      let user_executable = 0o100 & stat.mode() == 0o100;
      if user_executable {
```

However, I see that's exactly the logic which is implemented by the `tar` crate here:

https://github.com/alexcrichton/tar-rs/blob/34744459084c1fffb03d6c742f5a5af9a6403bc4/src/header.rs#L765-L771

So it might be even butter to simply use the [`Deterministic`](https://docs.rs/tar/0.4.40/tar/enum.HeaderMode.html#variant.Deterministic) header mode with something like

```rust
let mut tar_builder = tar::Builder::new(dest_file).mode(HeaderMode::Deterministic);
```

then use `append_dir` and `append_file` as before #7980.

The only difference is that mtime is not copied in the `Deterministic` mode and a default value is used, but I really don't know if it's a feature or a bug for us, and what the mtime should be in .deb packages.

Reply:
Yeah, bitwise would be better and more readable code. Thanks for the suggestion!

And regarding Header Mode, I will be refactoring the code.